### PR TITLE
EN-12743: account storage exporter

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
New tool that exports the storage of an account based on a certain DB, alongside with a root hash.

Heavily inspired by the `trieChecker` app

### CODE REVIEW

File: accountStorageExporter/main.go [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+	if err != nil {
+		return err
+	}
+
+	keyValueMap := make(map[string]string)
+	for leaf := range leavesCh {
+		keyValueMap[hex.EncodeToString(leaf.Key())] = hex.EncodeToString(leaf.Value())
+	}
+
+	fmt.Println(keyValueMap)
```
Comment (iulianpascalau) [inline accountStorageExporter/main.go:237]: why not use the log subsystem?

File: accountStorageExporter/main.go [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+	leavesCh, err := userAccount.DataTrie().GetAllLeavesOnChannel(mainRootHash)
+	if err != nil {
+		return err
+	}
+
+	keyValueMap := make(map[string]string)
+	for leaf := range leavesCh {
+		keyValueMap[hex.EncodeToString(leaf.Key())] = hex.EncodeToString(leaf.Value())
+	}
+
```
Comment (iulianpascalau) [inline accountStorageExporter/main.go:236]: For consistency, I think we should sort the map. Subsequent calls can be easily diff checked.

File: accountStorageExporter/vars.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package main
```
Comment (iulianpascalau) [inline accountStorageExporter/vars.go:1]: there is a lot of boiler plate in this app: go/mof/go.sum, vars handling, the simplePathManagerComponent, and so on. We can refactor these by actually having only one project with multiple binaries. In this way we can share the common components

File: accountStorageExporter/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+- the AccountsTrie database from a snapshot
+- the State Root Hash used to recreate the trie
+- the Address to export the storage for
+
+
+## Output
+
+If everything is successfully, the map containing the key-value pairs from the address storage will be printed in the console and also written in the `output.json`
+
+Note that both the keys and the values and hex-encoded.
```
Comment (BeniaminDrasovean) [inline accountStorageExporter/README.md:17]: _are_ hex-encoded

File: accountStorageExporter/main.go (hunk 1 above)
Comment (bogdan-rosianu) [reply accountStorageExporter/main.go]: done

File: accountStorageExporter/main.go (hunk 2 above)
Comment (bogdan-rosianu) [reply accountStorageExporter/main.go]: Not a bad idea, but I cannot sort the map. I can only print it in a given order (by using an additional sorted slice for the keys), but I cannot guarantee the file's output. Also, I wouldn't change the output to array or something else

File: accountStorageExporter/vars.go (hunk 3 above)
Comment (bogdan-rosianu) [reply accountStorageExporter/vars.go]: done, restructured the entire repository

**Review by BeniaminDrasovean — APPROVED** [review]

File: trieTools/trieToolsCommon/vars.go [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package trieToolsCommon
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/hashing/blake2b"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+)
+
+var (
+	Hasher     = blake2b.NewBlake2b()
```
Comment (iulianpascalau) [inline trieTools/trieToolsCommon/vars.go:9]: missing comments

Comment (bogdan-rosianu) [reply trieTools/trieToolsCommon/vars.go]: done

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]

**Pull request merged**